biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -45,25 +46,28 @@ protected void connect(Class<Agent> agent, Supervisor supervisor, String host, i
 		connect(agent, supervisor, host, port, -1);
 	}
 
-	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, int timeout)
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, final int timeout)
```

Comment:
@bjhargrave  there is really two timeouts internally, there is the single socket connection timeout and also there is a total timeout for the overall agent connection.  I discussed with Andy and we didn't want to make this more complicated by introducing two timeout args and just leave it with one.  So while this code works, its a little weird to be sharing the timeout.  let us know if we should improve it, I'm going to merge for now.